### PR TITLE
Add UMD bundle for @xstate/react.

### PR DESCRIPTION
**Pull request by dimitardanailov**
The pull request is based on #1108. 

The pull request should close the following issue: https://github.com/davidkpiano/xstate/issues/1736

### CODE REVIEW

**Comment by davidkpiano** [comment]
Thanks @dimitardanailov! Can you also add a changeset? 

1. `yarn changeset`
2. Enter a changeset to make a `minor` change for `@xstate/react`, describing that you're adding UMD bundles.

**Comment by dimitardanailov** [comment]
@davidkpiano 

Please review my commit: https://github.com/davidkpiano/xstate/pull/1738/commits/b4f8dd4232b0567f218d8f53b438ffdd03d293f4

File: packages/xstate-react/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        '@xstate/react': 'XStateReact',
+        '@react/composition-api': 'reactCompositionApi'
+      }
+    },
+    plugins: [createTSCofig(), terser({ include: [/^.+\.min\.js$/] })]
+  };
+}
+
+export default [
+  createUmdConfig({
```
Comment (Andarist) [inline packages/xstate-react/rollup.config.js:32]: in v5 we won't be generating unminified UMD bundles, so let's drop this for now

File: packages/xstate-react/rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      file,
+      format: 'umd',
+      name,
+      globals: {
+        xstate: 'XState',
+        '@xstate/react': 'XStateReact',
+        '@react/composition-api': 'reactCompositionApi'
+      }
+    },
+    plugins: [createTSCofig(), terser({ include: [/^.+\.min\.js$/] })]
```
Comment (Andarist) [inline packages/xstate-react/rollup.config.js:27]: as mentioned in the other comment - let's just focus on producing minified bundles

File: packages/xstate-react/rollup.config.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  }),
+  createUmdConfig({
+    name: 'XStateReactFSM',
+    input: 'src/fsm.ts',
+    output: 'dist/xstate-react.fsm.js'
+  }),
+  createUmdConfig({
+    name: 'XStateReactFSM',
+    input: 'src/fsm.ts',
+    output: 'dist/xstate-react.fsm.min.js'
```
Comment (Andarist) [inline packages/xstate-react/rollup.config.js:50]: Similar to the other comment about v5 - we should be naming this file like this:
```suggestion
    output: 'dist/xstate-react-fsm.umd.min.js'
```

File: packages/xstate-react/README.md [hunk 4]
```diff
@@ -29,6 +29,22 @@
 npm i xstate @xstate/react
 ```
 
+**Via CDN**
+
+```html
+<script src="https://unpkg.com/@xstate/react/dist/xstate-react.min.js"></script>
```
Comment (Andarist) [inline packages/xstate-react/README.md:35]: after adjusting configs please remember about adjusting those URLs here

**Comment by dimitardanailov** [comment]
@Andarist 

Thank you very much for the additional advices. I've created three new commits. Please review my improvements.

**Comment by Andarist** [comment]
Thanks!

**Comment by dimitardanailov** [comment]
@Andarist and @davidkpiano 

Thank you very much for the collaboration. I'd like to share: What I found. Before that: I'd like to give you access to my playground area:

branch: `xstate-react-umd` -> https://github.com/dimitardanailov/react-components/tree/xstate-react-umd
Component: 
https://github.com/dimitardanailov/react-components/blob/xstate-react-umd/packages/d3-tree-collapsible/app/components/D3Tree.js

The component throws an error if the import is: 
```
const { useMachine } = XStateReact
```

The error is: `react-hot-loader.development.js:297 ReferenceError: process is not defined`

Screen Shot 2020-12-19 at 10 59 03 PM

Screen Shot 2020-12-19 at 11 05 11 PM

My recommendation is: https://www.npmjs.com/package/@rollup/plugin-replace
The package is able to replace string sections in the particular file. 

What do you think ?

**Comment by Andarist** [comment]
Ah - yes. This is the appropriate approach. I've missed that this will be required during the code review. Are you willing to prepare a PR with the fix?

**Comment by dimitardanailov** [comment]
@Andarist 

Please review my pull request: 
https://github.com/davidkpiano/xstate/pull/1756

**Pull request merged**